### DISCUSSION TITLE
Opportunity to expand possibilities of Mpesa Rust API

### DISCUSSION THREAD

**Discussion by kcelestinomaria** [collinsmuriuki/mpesa-rust#29] [Ideas]
Hello @collinsmuriuki , my name is Celestino, Kenyan as well, was wondering if we could try interfacing the Mpesa Rust API with several crypto libraries to see if we can create a free seamless transaction from Mpesa to crypto. There is a hackathon going on by a blockchain project called Solana, and they are giving prizes for Decentralized Finance projects, so I was seeing this as an opportunity to battle-test the Mpesa Rust API with crypto and Decentralized Finance, possibly implement a simple connector or on/off-ramp Mpesa service API in Rust. By the way, Solana is a Rust-specific blockchain project, so it will be easy to prototype as well. Thank you

**Comment by c12i** [comment]
Hey @kcelestinomaria, that would be great. I am aware of Solana but haven't gotten to interacting with the network. Would be interested to know what you have in mind with regards to execution.
